test/python/utils/test_deprecation.py

```diff
@@ -0,0 +1,157 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for the methods in ``utils.deprecation``."""
+
+
+from qiskit import QiskitError
+from qiskit.utils.deprecation import deprecate_function, deprecate_arguments
+from qiskit.test import QiskitTestCase
+
+
+class DummyClass:
+    """This is short description. Let's make it
+    multiline"""
+
+    def __init__(self, arg1: int = None, arg2: [int] = None):
+        self.arg1 = arg1
+        self.arg2 = arg2
+
+    @deprecate_function(
+        "The DummyClass.foo() method is being deprecated. " "Use the DummyClass.some_othermethod()",
+        docstring_version="1.2.3",
+    )
+    def foo_deprecated(self, index_arg2: int):
+        """A multi-line
+        docstring.
+
+        Here are more details.
+
+        Args:
+            index_arg2: `index_arg2` description
+
+        Returns:
+            int: returns `arg2[index_arg2]`
+
+        Raises:
+            QiskitError: if `len(self.arg2) < index_arg2`
+        """
+        if len(self.arg2) < index_arg2:
+            raise QiskitError("there is an error")
+        return self.arg2[index_arg2]
+
+    @deprecate_arguments({"if_arg1": "other_if_arg1"}, docstring_version="1.2.3")
+    def bar_with_deprecated_arg(
+        self, if_arg1: int = None, index_arg2: int = None, other_if_arg1: int = None
+    ):
+        """
+        A multi-line short
+        docstring.
+
+        This is the long description
+
+        Args:
+            if_arg1: `if_arg1` description with
+               multi-line
+            index_arg2: `index_arg2` description
+            other_if_arg1: `other_if_arg1` description
```

Comment:
We should add a test when one or more of the arguments has multiple paragraphs to describe it (or Sphinx block-level constructs).